terraform-modules/storage-bucket/bucket.tf

```diff
@@ -0,0 +1,54 @@
+# This module creates a single bucket
+
+resource "google_storage_bucket" "bucket" {
+  name          = var.bucket_name
+  provider = google
+
+  location = var.location
+  storage_class = var.storage_class
+
+  versioning {
+    enabled = var.versioning
+  }
+
+  dynamic "retention_policy" {
+    for_each = var.retention_policy == null ? [] : [var.retention_policy]
+    content {
+      is_locked        = var.retention_policy.is_locked
+      retention_period = var.retention_policy.retention_period
+    }
+  }
+
+  dynamic "lifecycle_rule" {
```

Comment:
not sure from the code but is the default mode to have no lifecycle policy set.  so if I don't provide a lifecycle policy when I call the module - then lifecycle is not enabled on bucket.

Reply:
to further clarify does providing a empty map to terraform for lifecycle rule - mean lifecycle is disabled.  or is it enabled but with a do nothing clause?  easy to test - create a bucket not specifying lifecycle and then run gsutil lifecycle get on bucket.  that will return if it is enabled or not and what the rule is.

Reply:
I confirm when I am testing the terraform plan output provides this result with no lifecycle  block declared : 

# module.aou-in-out-bucket.google_storage_bucket.bucket will be created
  + resource "google_storage_bucket" "bucket" {
      + bucket_policy_only = (known after apply)
      + force_destroy      = false
      + id                 = (known after apply)
      + location           = "US"
      + name               = "aou-input"
      + project            = (known after apply)
      + self_link          = (known after apply)
      + storage_class      = "MULTI_REGIONAL"
      + url                = (known after apply)

      + versioning {
          + enabled = true
        }
    }

And when I ran the command : 
wm745-040:test ftraquet$ gsutil lifecycle get gs://aou-input
gs://aou-input/ has no lifecycle configuration.

Reply:
I then updated my test.tf file to include lifecycle and then updated the bucket.
Lifecycle is present 
wm745-040:test ftraquet$ gsutil lifecycle get gs://aou-input
gs://aou-input/ has no lifecycle configuration.